### PR TITLE
[OSD-7556] update muo deployment docs

### PR DESCRIPTION
**Pull request by T0MASD**
### What type of PR is this?
_documentation_

### What this PR does / why we need it?

- Update development docs to add section on updating MUO with steps for
  - Building and pushing image to a registry
  - Updating MUO on a cluster

### Which Jira/Github issue(s) this PR fixes?

_Fixes #OSD-7556_

#### Optional
- Build and push MUO image to quay (follow instructions on [setting up quay](https://github.com/openshift/managed-upgrade-operator/blob/OSD-7556/docs/quay.md))
```
operator-sdk build quay.io/<QUAY_USERNAME>/managed-upgrade-operator:latest
podman login -u="<QUAY_SERVICE_ACCOUNT>" -p="<QUAY_SERVICE_ACCOUNT_TOKEN>" quay.io 
podman push quay.io/<QUAY_USERNAME>/managed-upgrade-operator:latest
```
- Update image path in `test/deploy/operator.yaml`
```
sed -i 's/quay.io\/app-sre/quay.io\/<QUAY_USERNAME>/g' test/deploy/operator.yaml
```
#### Desired
- Create new cluster
```
export MY_CLUSTERNAME=
ocm create cluster ${MY_CLUSTERNAME} --version 4.7.16 --provider gcp --region australia-southeast1
```
- Login to the cluster
```
backplane-login ${MY_CLUSTERNAME}
```
- Login to oc
```
export MY_CLUSTERID=
ocm get /api/clusters_mgmt/v1/clusters/${MY_CLUSTERID}/credentials | jq -r .kubeconfig > /tmp/kubeconfig.txt
export KUBECONFIG=/tmp/kubeconfig.txt
```
- Scale down current muo
```
oc scale deployment managed-upgrade-operator -n openshift-managed-upgrade-operator --replicas=0
```
- Create new project
```
oc new-project test-managed-upgrade-operator
```
- Deploy resources, run upgrade
```
oc create -f deploy/cluster_role.yaml
oc create -f test/deploy/cluster_role_binding.yaml
oc create -f test/deploy/service_account.yaml
oc create -f test/deploy/managed-upgrade-operator-config.yaml
oc create -f test/deploy/operator.yaml
oc create -f test/deploy/upgrade.managed.openshift.io_v1alpha1_upgradeconfig_cr.yaml
```
- Wait for deploy to finish
```
oc get clusterversion -w
```

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [x] Ran `make generate` command locally to validate code changes
- [x] Included documentation changes with PR

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -92,84 +93,143 @@ $ boilerplate/_lib/container-make
...
-$ oc new-project managed-upgrade-operator
+$ operator-sdk build quay.io/<QUAY_USERNAME>/managed-upgrade-operator:latest
 ```
 
-- Build the image. In this example, we will use [Quay](http://quay.io/) as the container registry for our image.
+- Push the image to the registry:
+
+---
+
+**NOTE**
```
Comment (bmeng) [inline docs/development.md:140]: The note here split the topic into two parts, makes me a bit confusing when reading.

Maybe put the note to the bottom of the whole Remotely topic?

File: docs/development.md [hunk 2]
```diff
@@ -92,84 +93,143 @@ $ boilerplate/_lib/container-make
...
 ```
 
-- Push the image to the registry.
+- Login to `oc` [as admin](https://github.com/openshift/ops-sop/blob/master/v4/howto/break-glass-kubeadmin.md#for-clusters-with-public-api) 
+  
+- If MUO is already deployed on the cluster in `openshift-managed-upgrade-operator` namespace, scale the deployment down to 0:
 
 ```shell
-podman push quay.io/myuser/managed-upgrade-operator:latest
+oc scale deployment managed-upgrade-operator -n openshift-managed-upgrade-operator --replicas=0
```
Comment (mrbarge) [inline docs/development.md:155]: This presumes other MUO users might already be running MUO in the `openshift-managed-upgrade-operator` namespace, which may not be correct (they may be running on a plain OCP cluster). 

Perhaps just a "ensure no other instances of `managed-upgrade-operator` are actively running on your cluster, as they may conflict" will suffice.

File: docs/development.md [hunk 3]
```diff
@@ -92,84 +93,143 @@ $ boilerplate/_lib/container-make
...
+Refer to [fast-4.7](https://access.redhat.com/labs/ocpupgradegraph/update_channel?channel=fast-4.7&arch=x86_64&is_show_hot_fix=false) for currently available versions in `fast-4.7` channel.
+
+---
+
+## Monitoring ongoing upgrade
+
+- After applying an `upgradeConfig`, you should see your upgrade progressing
+
+```shell
+oc get upgrade
```
Comment (mrbarge) [inline docs/development.md:205]: To have consistency with other included commands which are namespace-bound
```suggestion
oc get upgrade -n test-managed-upgrade-operator
```

File: docs/development.md [hunk 4]
```diff
@@ -92,84 +93,143 @@ $ boilerplate/_lib/container-make
...
+``` 
+
+- Inspect `upgradeConfig`:
+
+```shell
+oc describe upgrade managed-upgrade-config
```
Comment (mrbarge) [inline docs/development.md:211]:
```suggestion
oc describe upgrade -n test-managed-upgrade-operator managed-upgrade-config 
```

File: docs/quay.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#### Parepare quay repo
```
Comment (mrbarge) [inline docs/quay.md:1]:
```suggestion
#### Prepare quay repo
```

File: test/deploy/operator.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: node-role.kubernetes.io/master
+                operator: In
+                values:
+                - ""
+      tolerations:
```
Comment (mrbarge) [inline test/deploy/operator.yaml:26]: For a "test" deployment I don't think we need to include tolerations of these master taints, but that's a minor thing.

File: test/deploy/operator.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+      - effect: NoExecute
+        key: node-role.kubernetes.io/master
+      containers:
+        - name: managed-upgrade-operator
+          # Replace this with the built image name
+          # This will get replaced on deploy by /hack/generate-operator-bundle.py
+          image: quay.io/app-sre/managed-upgrade-operator:latest
```
Comment (mrbarge) [inline test/deploy/operator.yaml:35]: It is probably worth calling out in the documentation instructions that the person would need to modify this line to point to their personal repo,. as they may not notice the comment in the yaml.

File: test/deploy/operator.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      serviceAccountName: managed-upgrade-operator
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: node-role.kubernetes.io/master
+                operator: In
+                values:
+                - ""
```
Comment (mrbarge) [inline test/deploy/operator.yaml:25]: Sorry, I should have mentioned, this whole section can be removed as well. (in conjunction with removal of the toleration section)

**Comment by mrbarge** [comment]
Thanks for getting this all sorted out
/lgtm

**Pull request merged**